### PR TITLE
Recommend adding an API docs section

### PR DESCRIPTION
**Pull request by oandregal**
This PR adds a new recommendation for when we create a new package: adding a section about API docs using the tool we created to auto-generate them from JSDoc comments.

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ When creating a new package, you need to provide at least the following:
     - Package description
     - Installation details
     - Usage example
+    - API Docs, if applicable (auto-generated with `packages/docgen`).
```
[the remark this answers is not in the document]
Comment (nosolosw) [reply packages/README.md]: I just added one in this same file https://github.com/WordPress/gutenberg/pull/22415/commits/526c50d3f1b11ab7b0d0ee4c440664180a80db36

File: packages/README.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ When creating a new package, you need to provide at least the following:
     - Package description
     - Installation details
     - Usage example
+    - API Docs, if applicable. See the "Maintaining API docs" section.
```
Comment (chrisvanpatten) [inline packages/README.md:47]:
```suggestion
    - API documentation, if applicable (see the "Maintaining API docs" section)
```

File: packages/README.md [hunk 3]
```diff
@@ -112,6 +113,32 @@ npm install glob --save-dev
...
+<!-- START TOKEN(Autogenerated API docs) -->
+
+Content within the HTML comment will be replaced by the generated documentation.
+
+<!-- END TOKEN(Autogenerated API docs) -->`.
+```
+
+Each time there is a commit to the public API of the package the `README.md` will be updated and kept in sync.
+
+The above snippet within the package's `README.md` signals the Gutenberg utilities to go to `src/index.js` and extract the JSDoc comments of the export statements into a more friendly format. Packages may want to use a different source file or add the exports of many files into the same `README.md` (see `packages/core-data/README.md` as an example); and they can do so by adding the relative path to be used as source into the HTML comment:
```
Comment (chrisvanpatten) [inline packages/README.md:132]:
```suggestion
The above snippet within the package's `README.md` signals the Gutenberg utilities to go to `src/index.js` and extract the JSDoc comments of the export statements into a more friendly format.

Packages may want to use a different source file or add the exports of many files into the same `README.md` (see `packages/core-data/README.md` as an example); they can do so by adding the relative path to be used as source into the HTML comment:
```

**Review by chrisvanpatten — CHANGES_REQUESTED** [review]
Two small nits then I think this is good to 🚢

File: packages/README.md (hunk 2 above)
Comment (gziolo) [reply packages/README.md:47]: Ideally, "Maintaining API docs" would link to `#maintaining-api-docs`.

**Review by gziolo — COMMENTED** [review]
Nice, I like all the details included. I will leave the final ✅ to @chrisvanpatten. It looks great from my perspective 💯

**Review by gziolo — APPROVED** [review]

**Review by gziolo — APPROVED** [review]

**Comment by nosolosw** [comment]
@chrisvanpatten it looks like this is blocked until your approval. Hopefully, all your comments were addressed and you can give a quick :heavy_check_mark: without taking too much of your time?

**Comment by gziolo** [comment]
I unblocked this PR 😃 @chrisvanpatten let us know if there is anything left whenever it fits you and thanks for valuable  feedback as always  😃

**Comment by chrisvanpatten** [comment]
@nosolosw @gziolo Sorry I missed the ping — had a couple crazy days at work. Definitely never hesitate to dismiss my review!

**Pull request merged**